google-cloud-dialogflow-cx/src/test/java/com/google/cloud/dialogflow/cx/v3beta1/it/ITSystemTest.java

```diff
@@ -0,0 +1,609 @@
+            .setDescription(DESCRIPTION)
+            .setTimeZone(AGENT_TIME_ZONE)
+            .setDefaultLanguageCode(DEFAULT_LANGUAGE_CODE)
+            .setEnableStackdriverLogging(true)
+            .setEnableSpellCorrection(true)
+            .build();
+    Agent response = agentsClient.createAgent(PARENT, agent);
+    String agentName = response.getName();
+    try {
+      /* Replaces the current agent with a new one. Note that all existing resources in agent (e.g.
+       * intents, entity types, flows) will be removed
+       */
+      RestoreAgentRequest restoreAgentRequest =
+          RestoreAgentRequest.newBuilder().setName(agentName).build();
+      OperationFuture<Empty, Struct> operationFuture =
+          agentsClient.restoreAgentAsync(restoreAgentRequest);
+      assertNotEquals(agentName, operationFuture.getName());
+    } finally {
+      DeleteAgentRequest deleteAgentRequest =
+          DeleteAgentRequest.newBuilder().setName(agentName).build();
+      agentsClient.deleteAgent(deleteAgentRequest);
+    }
+  }
+
+  @Test
+  public void getEntityTypeTest() {
+    GetEntityTypeRequest request =
+        GetEntityTypeRequest.newBuilder().setName(entityTypesName).build();
+    EntityType entityType = entityTypesClient.getEntityType(request);
+    assertEntityTypesDetails(entityType);
+  }
+
+  @Test
+  public void listEntityTypesTest() {
+    ListEntityTypesRequest request =
+        ListEntityTypesRequest.newBuilder().setParent(agentName).build();
+    Iterable<EntityType> entityTypes = entityTypesClient.listEntityTypes(request).iterateAll();
+    for (EntityType entityType : entityTypes) {
+      if (entityTypesName.equals(entityType.getName())) {
+        assertEntityTypesDetails(entityType);
+      }
+    }
+  }
+
+  @Test
+  public void updateEntityTypeTest() {
+    EntityType entityType =
+        EntityType.newBuilder()
+            .setName(entityTypesName)
+            .setKind(EntityType.Kind.KIND_LIST)
+            .setDisplayName(DISPLAY_NAME)
+            .setKindValue(2)
+            .build();
+    UpdateEntityTypeRequest updateEntityTypeRequest =
+        UpdateEntityTypeRequest.newBuilder().setEntityType(entityType).build();
+    EntityType response = entityTypesClient.updateEntityType(updateEntityTypeRequest);
+    assertEquals(2, response.getKindValue());
+  }
+
+  @Test
+  public void getFlowTest() {
+    GetFlowRequest request = GetFlowRequest.newBuilder().setName(flowName).build();
+    Flow flow = flowsClient.getFlow(request);
+    assertFlowDetails(flow);
+  }
+
+  @Test
+  public void listFlowsTest() {
+    ListFlowsRequest request = ListFlowsRequest.newBuilder().setParent(agentName).build();
+    FlowsClient.ListFlowsPagedResponse pagedListResponse = flowsClient.listFlows(request);
+    List<Flow> flows = Lists.newArrayList(pagedListResponse.iterateAll());
+    for (Flow flow : flows) {
+      if (flow.getName().equals(flowName)) {
+        assertFlowDetails(flow);
+      }
+    }
```

Comment:
if flow.getName() never returns flowName the test will pass.
You need to add a flag that assertFlowDetails() has been called and fail the test if this flag is false.

Reply:
done